### PR TITLE
Change hardcoded branch master to default branch

### PR DESCRIPTION
**Pull request by gilMars**
Changes to make ml-git call the default branch of a metadata repository, instead of the master branch.

### CODE REVIEW

**Review by Mun1z — APPROVED** [review]

File: tests/integration/test_08_checkout_tag.py [hunk 1]
```diff
@@ -340,3 +340,12 @@ def test_24_check_sampling_flag_after_checkout(self):
...
         self.assertNotIn(ERROR_MESSAGE, check_output(MLGIT_CHECKOUT % ('dataset', 'computer-vision__images__dataset-ex__2')))
         self.assertFalse(self.check_sampling_flag('dataset'))
+
+    @pytest.mark.usefixtures('start_local_git_server_with_main_branch', 'switch_to_tmp_dir')
+    def test_25_checkout_tag_with_main_branch(self):
+        self.set_up_checkout('dataset')
+        check_output(MLGIT_CHECKOUT % ('dataset', 'computer-vision__images__dataset-ex__1'))
+        file = os.path.join(self.tmp_dir, 'dataset', 'computer-vision', 'images', 'dataset-ex', 'newfile0')
+        self.check_metadata()
+        self.check_amount_of_files('dataset', 6, sampling=False)
```
Comment (andersonpablosilva) [inline tests/integration/test_08_checkout_tag.py:350]: Can we use variables with meaningful names for this numbers? Ex:
`self.check_amount_of_files('dataset', total_files_in_repository, sampling=False)`

Comment (gilMars) [reply tests/integration/test_08_checkout_tag.py]: Adjusted in 384720a8.

**Review by andersonpablosilva — APPROVED** [review]

File: ml_git/_metadata.py [hunk 2]
```diff
@@ -72,10 +72,26 @@ def check_exists(self):
...
         return True
 
-    def checkout(self, sha):
+    def checkout(self, sha=None):
         repo = Git(self.__path)
+        if sha is None:
+            sha = self.get_default_branch()
         repo.checkout(sha)
 
+    def get_symbolic_ref(self):
```
Comment (diego-pedroso) [inline ml_git/_metadata.py:81]: Is this supposed to be a private function? If so, please add a leading underscore.

Comment (gilMars) [reply ml_git/_metadata.py]: Added leading underscore in method signature to determine an internal use. 661b293

**Review by diego-pedroso — APPROVED** [review]

**Pull request merged**